application/aam-backend-service/src/test/kotlin/com/aamdigital/aambackendservice/export/usecase/DefaultRenderTemplateUseCaseTest.kt

```diff
@@ -0,0 +1,380 @@
+            templateStorage = templateStorage,
+        )
+    }
+
+    @Test
+    fun `should return Failure when fetchTemplate returns an error`() {
+        // Arrange
+        val templateRef = DomainReference("some-id")
+        val bodyData: JsonNode = TextNode("body-data")
+        val exception = InternalServerException(message = "fetchTemplate error", cause = null)
+
+        whenever(templateStorage.fetchTemplate(templateRef)).thenReturn(Mono.error(exception))
+
+        // Act & Assert
+        StepVerifier.create(
+            service.execute(
+                CreateRenderTemplateRequest(
+                    templateRef, bodyData
+                )
+            )
+        ).assertNext {
+            assertThat(it).isInstanceOf(UseCaseOutcome.Failure::class.java)
+            assertEquals(
+                CreateRenderTemplateErrorCode.FETCH_TEMPLATE_FAILED_ERROR, (it as UseCaseOutcome.Failure).errorCode
+            )
+        }.verifyComplete()
+    }
+
+    @Test
+    fun `should return Failure when json response could not be parsed`() {
+        // given
+        val templateRef = DomainReference("some-id")
+        val bodyData: JsonNode = TextNode("body-data")
+        val exportTemplate = ExportTemplate(
+            id = "export-id",
+            templateId = "export-template-id",
+            title = "export-title",
+            description = "export-description",
+            applicableForEntityTypes = emptyList()
+        )
+
+        whenever(templateStorage.fetchTemplate(templateRef)).thenReturn(Mono.just(exportTemplate))
+        mockWebServer.enqueue(MockResponse().setBody("invalid json"))
+
+        // when
+        StepVerifier.create(
+            service.execute(
+                CreateRenderTemplateRequest(
+                    templateRef, bodyData
+                )
+            )
+        ).assertNext {
+            // then
+            assertThat(it).isInstanceOf(UseCaseOutcome.Failure::class.java)
+            assertEquals(
+                CreateRenderTemplateErrorCode.PARSE_RESPONSE_ERROR, (it as UseCaseOutcome.Failure).errorCode
+            )
+        }.verifyComplete()
+    }
+
+    @Test
+    fun `should return Failure when fetchTemplateRequest throws exception`() {
+        // given
+        val templateRef = DomainReference("some-id")
+        val bodyData: JsonNode = TextNode("body-data")
+
+        whenever(templateStorage.fetchTemplate(templateRef)).thenAnswer {
+            throw InvalidArgumentException()
+        }
+
+        // when
+        StepVerifier.create(
+            service.execute(
+                CreateRenderTemplateRequest(
+                    templateRef, bodyData
+                )
+            )
+        ).assertNext {
+            // then
+            assertThat(it).isInstanceOf(UseCaseOutcome.Failure::class.java)
+            assertEquals(
+                CreateRenderTemplateErrorCode.INTERNAL_SERVER_ERROR, (it as UseCaseOutcome.Failure).errorCode
```

Comment:
would this not rather be an 40x error (user sent invalid templateRef in request)?